### ISSUE TITLE
`Sender.Path` incorrectly identified as `deadLetters` after upgrading from Akka 1.0.8 --> 1.3.8

### ISSUE THREAD

**Issue by SeanKilleen** [akkadotnet/akka.net#3566]
Hi all,

### Background

As part of an Akka intro talk I've got a sample app that writes out to a `ConsoleWriterActor`. The console writer actor is part of a demo, and uses `Sender.Path` in the console output to show where the message came from.

I was prepping a new demo and realized the console was outputting that the `Sender.Path` was `deadLetters`, which seemed odd. I checked my older app and it worked as expected. I then started looking for what I changed.

As far as I can tell, supported by BeyondCompare, all I've changed is:

* .NET Target: I changed the apps to target 4.6.1 (from 4.5)
* Packages: I upgraded to Akka.NET 1.3.8 and the latest stable version of all other packages. (it may be in one of these other packages, e.g. Json.NET, where I've introduced an incompatibility that I'm not seeing).

### Sample app to Reproduce

* https://github.com/SeanKilleen/Presentations
* `cotb2018` branch (https://github.com/SeanKilleen/Presentations/tree/cotb2018)
* `2018-04_AkkaIntro` and `2018-08_AkkaIntro` folders each have a `code` folder that contains the sample app.
* Open the solution and run the `Routing` console app as-is. 

### Other Information

* OS: Windows 10 x64 Version 1803 (OS Buils 17134.165)

**Comment by SeanKilleen** [comment]
Working to narrow it down. Will revert & try baby steps to see where it goes wrong.

**Comment by SeanKilleen** [comment]
OK, I seem to have narrowed the repro steps down.

✅ Open the project as-is. Behavior verified.
✅ Update target frameworks from `4.5` to `4.6.1`. Still works as expected.
❓ Update Akka 1.0.8 --> 1.3.8 Project fails at runtime due to missing Akka.Remote update.
❌ Update Akka.Remote and associated dependencies. Behavior begins at this point.

**Comment by SeanKilleen** [comment]
OK, I think this is actually expected behavior. 

The issue is that I'm sending the message from my app itself, not from an actor within the actor system.

* The prior behavior shows `Sender.Path` as `akka://all-systems/`, which seems like an early placeholder. 
* The current behavior shows `Sender.Path` as `akka://ElasticSystem/deadLetters`.

This makes sense. It's coming from within my actor system, but there's no actual reference, so it's sent to dead letters.

I only saw it because of the deadLetters designation but it's obviously correct. 

Closing. Thaaaanks for playing!
